easy_study/lib/presenter/TimeTracking.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:async';
+
+import 'package:easy_study/Database/DBHelper.dart';
+import 'package:easy_study/model/Subject.dart';
+import 'package:flutter/material.dart';
+
+class TimeTracking extends StatefulWidget {
+  final Subject subject;
+
+  const TimeTracking({Key key, this.subject}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _TimeTracking();
+}
+
+class _TimeTracking extends State<TimeTracking> {
+  Timer timer;
+
+  String printedTime = '00:00:00';
+  DateTime elapsedTime;
+
+  startTimer() {
+    elapsedTime = DateTime.parse(widget.subject.startTimetracking);
+    printedTime = getDifference(elapsedTime);
+    if (timer == null || !timer.isActive) {
+      timer = new Timer.periodic(new Duration(milliseconds: 1000), updateTime);
+    }
+  }
+
+  updateTime(Timer timer) {
+    setState(() {
+      printedTime = getDifference(elapsedTime);
+    });
+  }
+
+  startWatch() {
+    if (timer == null || !timer.isActive) {
+      elapsedTime = DateTime.now();
+      DBHelper().updateSingleString(widget.subject.id,
```

Comment:
Sure we have a singleton DB helper, but our architecture wants us to change the state, not the DB directly.
```suggestion
new StoreConnector<AppState, Store>(
      converter: (store) => store,
      builder: (context, callback) {
      callback..dispatch(UpdateSubject(*yourSubject*))
      }
```
U have to update the subject. Maybe test it first.